core/amber/src/main/python/core/storage/iceberg/iceberg_document.py

```diff
@@ -0,0 +1,259 @@
+                except Exception:
+                    print("Could not read iceberg table:\n")
+                    raise Exception
+            else:
+                return iter([])
+
+    def _extract_sorted_file_scan_tasks(self, current_snapshot):
+        """
+        As self.table.inspect.entries() does not work with java files, this method
+        implements the logic to find file_sequence_number for each data file ourselves
+        :param current_snapshot: The current snapshot of the table.
+        :return: The file scan tasks of the file sorted by file_sequence_number
+        """
+        file_sequence_map = {}
+        for manifest in current_snapshot.manifests(self.table.io):
+            for entry in manifest.fetch_manifest_entry(io=self.table.io):
+                file_sequence_map[entry.data_file.file_path] = entry.sequence_number
+        # Retrieve and sort the file scan tasks by file sequence number
+        file_scan_tasks = list(self.table.scan().plan_files())
+        # Sort files by their sequence number. Files without a sequence
+        # number will be read last.
+        sorted_file_scan_tasks = sorted(
+            file_scan_tasks,
+            key=lambda t: file_sequence_map.get(t.file.file_path, float("inf")),
+        )
+        return sorted_file_scan_tasks
+
+    def __iter__(self) -> Iterator[T]:
+        return self
+
+    def __next__(self) -> T:
+        if self.num_of_returned_records >= self.total_records_to_return:
+            raise StopIteration("No more records available")
+
+        while True:
+            try:
+                record = next(self.current_record_iterator)
+                self.num_of_returned_records += 1
+                return record
+            except StopIteration:
+                # current_record_iterator is exhausted, need to go to the next file
+                try:
+                    next_file = next(self.usable_file_iterator)
+                    arrow_table = read_data_file_as_arrow_table(next_file, self.table)
+                    self.current_record_iterator = self.deserde(
+                        self.table_schema, arrow_table
+                    )
+                    # Skip records within the file if necessary
+                    records_to_skip_in_file = (
+                        self.from_index - self.num_of_skipped_records
+                    )
+                    if records_to_skip_in_file > 0:
+                        self.current_record_iterator = self._skip_records(
+                            self.current_record_iterator, records_to_skip_in_file
+                        )
+                        self.num_of_skipped_records += records_to_skip_in_file
+                except StopIteration:
+                    # no more files left in this table
+                    raise StopIteration("No more records available")
+
+    @staticmethod
+    def _skip_records(iterator, count):
+        return iter(list(iterator)[count:])
```

Comment:
to avoid unnecessary copy, I suggest change this to:
```
from itertools import islice

def _skip_records(iterator, count):
    return islice(iterator, count, None)
```